### PR TITLE
Add Metafield Resource

### PR DESCRIPTION
**Pull request by aryy**
* Adds the general Metafield resource (e.g. `client.Metafield.List(...)`)
* Adds metafields to Product resource (e.g. `client.Product.ListMetafields(...)`)

### CODE REVIEW

File: goshopify.go [hunk 1]
```diff
@@ -201,11 +203,19 @@ func (c *Client) Do(req *http.Request, v interface{}) error {
 		return err
 	}
 
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
```
Comment (dlebech) [inline goshopify.go]: Why did you make changes to this file? This seems to be unrelated to the main PR, so could you revert it for now?

**Review by dlebech — CHANGES_REQUESTED** [review]
Thanks for the PR @aryy ! Looks very useful. I only have one comment on a piece of code that probably shouldn't be changed, and if you could fix the Travis error, that would be great. It's a small format problem in util_test and only seems to be a problem on Golang 1.10. The [error from Travis](https://travis-ci.org/getconversio/go-shopify/jobs/385965742):
```
./util_test.go:79: Errorf format %s has arg c.resourceID of wrong type int
```

**Comment by gir** [comment]
Sorry, we'll point the PR at another branch and open another PR for the order-related changes as well.

**Comment by aryy** [comment]
Closing in favor of https://github.com/getconversio/go-shopify/pull/90. This PR inadvertently pointed to our master branch.

**Pull request closed without merging**